devito/operator.py

```diff
@@ -188,6 +192,36 @@ def __init__(self, expressions, **kwargs):
 
     # Compilation
 
+    def _add_implicit(self, expressions):
+        """
+        Create and add to the expression tree any associated implicit
+        expressions.
+        """
+        updated_expressions = []
+        for e in expressions:
+            subdomain = e.subdomain
+            if bool(subdomain):
+                try:
+                    dat = e._subdomain._ie_dat
+                except AttributeError:
+                    dat = None
+                if bool(dat):
+                    dims = [d.root for d in e.free_symbols if isinstance(d, Dimension)]
+                    sub_dims = [d.root for d in subdomain.dimensions]
+                    dims = list(set(dims).symmetric_difference(set(sub_dims)))
```

Comment:
maybe you can use ``filter_ordered`` ? (``git grep filter_ordered`` for example usage) 

``filter_ordered`` will retain the ordering, while this way you're losing it due to casting to set (don't know if it's a potential issue or not)

Reply:
Ok thanks, will check.

Reply:
`filter_ordered` wont work here (unless I'm missing some mode it has) since we want the non-overlapping intersection of the two sets.

Order is not important (for the time being at least).

Reply:
> wdym ?

Currently, will anyone be using this functionality for Functions not of the form `F(t,x,y,z)`? In this case `(x,y,z)` will always be showing up in the subdomain and therefore we only need to filter the `t`. Order only becomes important when we have a function of the form `F(t1,t2,x,y,z)` and the order of `t1, t2` is important for loop construction. But anyway, I'll modify the code so that it should still work in the latter case.

> can you please remind me why do you have to compute the set difference of dims and subdims? I'm pretty sure 
> you've explained it to me, but I forgot

So in order to get the correct loop structure the implicit dims of the implicit equations needs to be (for say `F(t,xi,yi)`)
needs to be `dims(F)-dims(subdomain)+n` (i.e. `(t,n)`). Then the implicit dims of `F` need to be `(t,n,xi,yi)`. Hence the difference, then the addition of `n` and then later the addition of `subdomain.dimensions`. Does that make sense?

Reply:
yes, thanks, I see what you mean.